ui/selectors/gator-permissions/gator-permissions.ts

```diff
@@ -414,6 +492,69 @@ export const getAggregatedGatorPermissionByChainId = createSelector(
   },
 );
 
+/**
+ * Get aggregated list of gator permissions for a specific chainId and origin.
+ *
+ * @param _state - The current state
+ * @param options - The options to get permissions for
+ * @param options.aggregatedPermissionType - The aggregated permission type (e.g. 'token-transfer')
+ * @param options.chainId - The chainId to get permissions for (e.g. 0x1)
+ * @param options.siteOrigin - The site origin to filter by (e.g., 'https://example.com')
+ * @returns A aggregated list of gator permissions filtered by chainId and origin.
+ */
+export const getAggregatedGatorPermissionByChainIdAndOrigin = createSelector(
+  [
+    getGatorPermissionsMap,
+    (
+      _state: AppState,
+      options: {
+        aggregatedPermissionType: string;
+        chainId: Hex;
+        siteOrigin: string;
+      },
+    ) => options,
+  ],
+  (
+    gatorPermissionsMap,
+    { aggregatedPermissionType, chainId, siteOrigin },
+  ): StoredGatorPermissionSanitized<Signer, PermissionTypesWithCustom>[] => {
+    switch (aggregatedPermissionType) {
+      case 'token-transfer': {
+        const nativeTokenStreams =
+          gatorPermissionsMap['native-token-stream'][chainId] || [];
+        const erc20TokenStreams =
+          gatorPermissionsMap['erc20-token-stream'][chainId] || [];
+        const nativeTokenPeriodicPermissions =
+          gatorPermissionsMap['native-token-periodic'][chainId] || [];
+        const erc20TokenPeriodicPermissions =
+          gatorPermissionsMap['erc20-token-periodic'][chainId] || [];
+
+        const allPermissions = [
+          ...nativeTokenStreams,
+          ...erc20TokenStreams,
+          ...nativeTokenPeriodicPermissions,
+          ...erc20TokenPeriodicPermissions,
+        ];
+
+        // Filter by origin
+        const decodedSiteOrigin = decodeURIComponent(siteOrigin);
```

Comment:
`decodeURIComponent()` may throw on malformed input. Do we need a safe-decode guard here?